tests/backup/backup_rbac_test.go

```diff
@@ -499,3 +498,434 @@ var _ = Describe("{VerifyRBACforInfraAdmin}", func() {
+			clusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, nonAdminCtx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			userClusterMap[appAdminUser] = make(map[string]string)
+			for _, clusterName := range []string{SourceClusterName, destinationClusterName} {
+				userClusterUID, err := Inst().Backup.GetClusterUID(nonAdminCtx, orgID, clusterName)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", clusterName))
+				userClusterMap[appAdminUser][clusterName] = userClusterUID
+			}
+		})
+		Step(fmt.Sprintf("Take backup of applications from the App-admin user %s", appAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking backup of applications from the app-admin user %s", appAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(appAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appAdminUser)
+			userScheduleName := fmt.Sprintf("backup-schedule-%v", RandomString(5))
+			scheduleNameMap[appAdminUser] = userScheduleName
+			scheduleBackupName, err := CreateScheduleBackupWithValidation(nonAdminCtx, userScheduleName, SourceClusterName, appAdminBackupLocationName, appAdminBackupLocationUID, scheduledAppContexts, make(map[string]string), orgID, preRuleNameMap[appAdminUser], preRuleUidMap[appAdminUser], postRuleNameMap[appAdminUser], postRuleUidMap[appAdminUser], periodicSchedulePolicyNameMap[appAdminUser], periodicSchedulePolicyUidMap[appAdminUser])
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation and validation of schedule backup with schedule name [%s]", userScheduleName))
+			userBackupNamesMap[appAdminUser] = SafeAppend(&mutex, userBackupNamesMap[appAdminUser], scheduleBackupName).([]string)
+		})
+
+		Step(fmt.Sprintf("Take restore of applications from the App-admin user %s", appAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking restore of applications from the App-admin user %s", appAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(appAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appAdminUser)
+			restoreNameMap[appAdminUser] = fmt.Sprintf("%s-%s", restoreNamePrefix, userBackupNamesMap[appAdminUser][0])
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, bkpNamespaces)
+			err = CreateRestoreWithValidation(nonAdminCtx, restoreNameMap[appAdminUser], userBackupNamesMap[appAdminUser][0], make(map[string]string), make(map[string]string), destinationClusterName, orgID, appContextsToBackup)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreNameMap[appAdminUser], userBackupNamesMap[appAdminUser][0]))
+		})
+
+		Step(fmt.Sprintf("Delete app-admin user %s backups from the user context", appAdminUser), func() {
```

Comment:
Can we use a loop here?

Reply:
here we are deleting only one backup from user.